File: 206-ReverseLinkedList-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        stack<int> s;
        ListNode* node=head; 
        while (node) {
            s.push(node->val);
            node=node->next;
        }
        node=head;
        while (node) {
            node->val=s.top();
            s.pop();
            node=node->next;
        }
        return head;
    }
};

File: 206-ReverseLinkedList-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode reverseList(ListNode head) {
        if (head==null||head.next==null) return head;
        Stack<Integer> stack=new Stack<>();
        ListNode node=head;
        while (node!=null) {
            stack.push(node.val);
            node=node.next;
        }
        node=head;
        while (node!=null) {
            node.val=stack.pop();
            node=node.next;
        }
        return head;
    }
}
